test/auth/auth-manager-test.js

```diff
@@ -0,0 +1,129 @@
+const mockgoose = require('mockgoose');
+const mongoose = require('mongoose');
+const bluebird = require('bluebird');
+const chai = require('chai');
+const chaiMoment = require('chai-moment');
+const jwt = require('jsonwebtoken');
+const authManager = require('../../src/auth/auth-manager');
+const authModels = require('../../src/auth/auth-models');
+const userManager = require('../../src/users/user-manager');
+
+const expect = chai.expect;
+const Session = authModels.Session;
+
+chai.use(chaiMoment);
+mockgoose(mongoose);
+
+describe('authManager', () => {
+  // Connect to the database.
+  before((done) => {
+    process.env.SECRET = 'testsecret';
+
+    mongoose.Promise = bluebird;
+    mongoose.connect('', done);
+  });
+
+  // Clear the database before each test case.
+  beforeEach((done) => {
+    mockgoose.reset();
+    done();
+  });
+
+  after((done) => { mongoose.unmock(done); });
+
+  describe('#login', () => {
+    it('should pass a student token to the callback.', (done) => {
+      const student = {
+        email: 'student@test.com',
+        password: 'P@ssw0rd!Student',
+        firstName: 'Test',
+        lastName: 'User',
+        role: 'Student',
+        studentId: '',
+      };
+
+      userManager.createUser(student, (createErr, uid) => {
+        expect(createErr).to.be.null;
+        expect(uid).to.be.a('string');
+
+        authManager.login(student.email, student.password, (loginErr, token) => {
```

Comment:
Isn't this also creating a new session?  This should check that a new session is created, as well. (There should be test cases for when a session already exists, as well).